### PR TITLE
Add events 4720 4722 4723 4724 4725 4726 4738 4740 4767 4781 - LogonID correlation

### PR DESCRIPTION
**Pull request by janniten**
Added User management events

Event | Description
-- | --
4720 | A user account was created
4722 | A user account was enabled
4723 | An attempt was made to change an account's password
4724 | An attempt was made to reset an account's password
4725 | An user account was disabled
4726 | An user account was deleted
4738 | An user account was changed
4740 | An user account was locked out
4767 | An account was unlocked
4781 | The name of an account was changed

The function addLogonIds in order to populate winlog.logon.id was remove and replaced by two functions copySubjectUserLogonId and copyTargettUserLogonId so we can put only the value that has sense/meaning in the context of the event and use that value to correlate.
(https://discuss.elastic.co/t/winlogbeat-new-ecs-fields-and-security-module-questions/190479/13)
I have also added a function to add the event.action for this events
image50%

### CODE REVIEW

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js [hunk 1]
```diff
@@ -19,6 +19,22 @@ var security = (function () {
...
+        "4720": "Account Created",
+        "4722": "Account Enabled",
+        "4723": "Password Change",
+        "4724": "Password Reset",
+        "4725": "Account Disabled",
+        "4726": "Account Deleted",
+        "4738": "Account Changed",
+        "4740": "Account Locked Out",
+        "4767": "Account Unlocked",
+        "4781": "Account Renamed"
```
Comment (janniten) [inline x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: "4781": "Account Renamed",

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js [hunk 2]
```diff
@@ -1030,6 +1046,15 @@ var security = (function () {
         return msobjsMessageTable[code];
     };
 
+    var addActionDesc = function(evt){
+        var code = evt.Get("event.code");
+        if (!code) {
+            return;
+        }
+        var eventActionDescription=eventActionTypes[code];
+        evt.Put("event.action",eventActionDescription)
```
Comment (janniten) [inline x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: evt.Put("event.action",eventActionDescription);

**Comment by janniten** [comment]
Travis CI failed.
It is not clear for me if the problem is in my code

make[1]: *** [system-tests-environment] Error 2
make[1]: Leaving directory `/home/travis/gopath/src/github.com/elastic/beats/filebeat'
make: *** [testsuite] Error 2
make: Leaving directory `/home/travis/gopath/src/github.com/elastic/beats/filebeat'
The command "make $TARGETS" exited with 2.

Any help will be appreciated

**Comment by andrewkroh** [comment]
jenkins, test it

You're PR probably didn't cause the Travis failure since it doesn't test Windows. I'll trigger the Jenkins testing which will test Winlogbeat on Windows.

**Comment by andrewkroh** [comment]
jenkins, test this

**Comment by adriansr** [comment]
jenkins, test this

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js [hunk 3]
```diff
@@ -1206,6 +1271,36 @@ var security = (function () {
         // 4672 - Special privileges assigned to new logon.
         4672: event4672.Run,
 
+        // 4720 - A user account was created
+		4720: userMgmtEvts.Run,
```
Comment (andrewkroh) [inline x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: Looks like a spacing issue due to tabs. Let's replace all the tabs with spaces.

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js [hunk 4]
```diff
@@ -1030,6 +1046,15 @@ var security = (function () {
         return msobjsMessageTable[code];
     };
 
+    var addActionDesc = function(evt){
+        var code = evt.Get("event.code");
+        if (!code) {
+            return;
+        }
+        var eventActionDescription=eventActionTypes[code];
```
Comment (andrewkroh) [inline x-pack/winlogbeat/module/security/config/winlogbeat-security.js]:
```suggestion
        var eventActionDescription = eventActionTypes[code];
```

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js (hunk 2 above)
Comment (andrewkroh) [inline x-pack/winlogbeat/module/security/config/winlogbeat-security.js]:
```suggestion
        evt.Put("event.action", eventActionDescription);
```

Comment (andrewkroh) [reply x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: To be defensive I would check that a valid action was returned from the event ID lookup. Like

```
if (eventActionDescription) {
    evt.Put("event.action", eventActionDescription);
}
```

**Comment by janniten** [comment]
> There are a few changes needed, but overall LGTM. Thanks! After you make the requested updates I can go ahead and run the `cd x-pack\winlogbeat\module\security` and `go test ./... -update` for you to regenerate the golden files for us to review.
> 
> Once Elastic Common Schema (ECS) defines a set of `event.category` values we'll come back to these modules and update them appropriately.

I have made the requested updates.  Please let me know If I should do something more.

**Comment by andrewkroh** [comment]
jenkins, test this

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js [hunk 5]
```diff
@@ -19,6 +19,22 @@ var security = (function () {
         "11": "CachedInteractive",
     };
 
+    var eventActionTypes = {
```
Comment (andrewkroh) [inline x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: @webmat This is adding these `event.action` values based on the event ID. Do you want these action values or would you like to hold off until the action values are more well defined?

Comment (janniten) [reply x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: @andrewkroh @webmat I was thinking about  that point this morning. Personally i think is is better to define the event.action as defined in for auditbeat (some of them can be maybe reused). For example: event.action -> deleted-group-account-from
what do you thing? I can modify the event.action in order to match or follow  the same criteria as auditbeat

Comment (janniten) [reply x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: I took the events actions from auditbeat'a logs and I think these event.action are more suitable.

        |"4624": "Logon Success"|"logged-in" |
        |"4625": "Logon Failed"|"logon-failed"|
        |"4634": "Logoff"|"logged-out"|
        |"4672": "Special Logon"|"logged-in-special"|
        |"4720": "Account Created"|"added-user-account"|
        |"4722": "Account Enabled"|"enabled-user-account"|
        |"4723": "Password Change"|"changed-password"|
        |"4724": "Password Reset"|"reseted-password"|
        |"4725": "Account Disabled"|"disabled-user-account"|
        |"4726": "Account Deleted"|"deleted-user-account"|
        |"4738": "Account Changed"|"modified-user-account"|
        |"4740": "Account Locked Out"|"locked-out-user-account"|
        |"4767": "Account Unlocked"|"unlocked-user-account"|
        |"4781": "Account Renamed"|"renamed-user-account"|

@andrewkroh @webmat  What do you think?

Comment (webmat) [reply x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: The field `event.action` is not reserved, and can therefore be defined each source. So no limitations coming from ECS in that regard :-)

These values for `event.action` look fine to me 👍

Comment (janniten) [reply x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: @andrewkroh do I change the event.actions to the ones in the table above or should I keep the previous ones?
I have already write the code to support the events for group management (**)  and some enhancements in order to  map the UAC values to a human readable form for events 4720 y 4738. If you prefer we can continue with this PR and once closed, I can create a new one with the new event.actions, new events support  and the UAC enhancements.
What do you think?

(*)
4727 (S): A security-enabled global group was created. 
4728 (S): A member was added to a security-enabled global group
4729 (S): A member was removed from a security-enabled global group
4730 (S): A security-enabled global group was deleted.
4731 (S): A security-enabled local group was created.
4732 (S): A member was added to a security-enabled local group.
4733 (S): A member was removed from a security-enabled local group.
4734 (S): A security-enabled local group was deleted.
4735 (S): A security-enabled local group was changed.
4737 (S): A security-enabled global group was changed.
4754 (S): A security-enabled universal group was created. 
4755 (S): A security-enabled universal group was changed. 
4756 (S): A member was added to a security-enabled universal group. 
4757 (S): A member was removed from a security-enabled universal group
4758 (S): A security-enabled universal group was deleted. 
4764 (S): A group’s type was changed.
4798 (S): A user's local group membership was enumerated.
4799 (S): A security-enabled local group membership was enumerated.

Comment (andrewkroh) [reply x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: I like the proposed ones that match up with auditbeat. Can you please change them in this PR? Then I'll push an update with the golden files.

Comment (janniten) [reply x-pack/winlogbeat/module/security/config/winlogbeat-security.js]: Done!

**Comment by andrewkroh** [comment]
jenkins, test this

**Comment by andrewkroh** [comment]
jenkins, test this

**Comment by andrewkroh** [comment]
jenkins, test this

**Pull request merged**